src/service/action-impl.js

```diff
@@ -86,8 +87,26 @@ export class ActionService {
     /** @const {!Window} */
     this.win = win;
 
+    /** @const @private {!Object<string, function(!ActionInvocation)>} */
+    this.actionHandlers_ = {};
+
+    /** @param {!Vsync} */
+    this.vsync_ = vsyncFor(this.win);
+
     // Add core events.
     this.addEvent('tap');
+
+    // Add core actions.
+    this.addAction('hide', invocation => {
```

Comment:
I thought about this for a bit. However, the only true way to do this with "hide" is to use a `position:fixed` overlay where we already prohibit ads. So I think "hide" is ok. "toggleClass" will be harder when we get to it from this point of view.